Server.MirForms/Account/PlayerInfoForm.cs

```diff
@@ -133,5 +133,122 @@ private void OpenAccountButton_Click(object sender, EventArgs e)
             form.ShowDialog();
         }
 
+        private void FlagSearchBox_TextChanged(object sender, EventArgs e)
+        {
+            TextBox flagTextBox = (TextBox)sender;
+
+            if (string.IsNullOrWhiteSpace(flagTextBox.Text))
+            {
+                ResultLabel.Text = string.Empty;
+                return;
+            }
+
+            if (int.TryParse(flagTextBox.Text, out int flagIndex))
+            {
+                if (flagIndex >= 0 && flagIndex < Character.Flags.Length)
+                {
+                    bool flagValue = Character.Flags[flagIndex];
+
+                    if (flagValue)
+                    {
+                        ResultLabel.Text = $"Flag {flagIndex} ACTIVE.";
+                        ResultLabel.ForeColor = Color.Green;
+                    }
+                    else
+                    {
+                        ResultLabel.Text = $"Flag {flagIndex} INACTIVE.";
+                        ResultLabel.ForeColor = Color.Red;
+                    }
+                }
+                else
+                {
+                    ResultLabel.Text = "Invalid flag index";
+                    ResultLabel.ForeColor = Color.Red;
+                }
+            }
+        }
+
+        private void FlagUp_Click(object sender, EventArgs e)
+        {
+            if (int.TryParse(FlagSearchBox.Text, out int currentValue))
+            {
+                int newValue = currentValue + 1;
+
+                FlagSearchBox.Text = newValue.ToString();
+            }
+            else
+            {
+                FlagSearchBox.Text = "1";
+            }
+        }
+
+        private void FlagDown_Click(object sender, EventArgs e)
+        {
+            if (int.TryParse(FlagSearchBox.Text, out int currentValue))
+            {
+                int newValue = currentValue - 1;
+
+                newValue = Math.Max(0, newValue);
+
+                FlagSearchBox.Text = newValue.ToString();
+            }
+        }
+
+        private void QuestSearchBox_TextChanged(object sender, EventArgs e)
+        {
+            TextBox questTextBox = (TextBox)sender;
+
+            if (string.IsNullOrWhiteSpace(questTextBox.Text))
+            {
+                QuestResultLabel.Text = string.Empty;
+                return;
+            }
+
+            if (int.TryParse(questTextBox.Text, out int questID))
+            {
+                if (questID < 1)
+                {
+                    QuestResultLabel.Text = "Invalid Quest ID";
+                    QuestResultLabel.ForeColor = Color.Red;
+                    return;
+                }
+
+                bool isQuestComplete = Character.CompletedQuests.Contains(questID);
+
+                QuestResultLabel.Text = $"Quest {questID} is {(isQuestComplete ? "Complete" : "Not complete")}";
```

Comment:
maybe needs to be abit more sophisticated for quest lookup. As well as completed also check active quests.

Reply:
Not sure how to change the code, I know that 0 is not complete and 1 is complete.